### ISSUE TITLE
Moving v3.0 to the server

### ISSUE THREAD

**Issue by mleblanc321** [WheatonCS/Lexos#473]
caleb and i worked a couple of hours to see if we could get v3.0 running on the server; after some snags, reinstalls, etc., we got 3.0 to work ... except the matplotlib stuff (e.g., fails on hierarchical dendrograms)

ERROR from LOGs
```
[Fri Jul 29 14:52:22 2016] [error] [client 155.47.45.34]     self.tk = _tkinter.create(screenName, baseName, className, interactive, wantobjects, useTk, sync, use), referer: http://lexos.wheatoncollege.edu/cluster
[Fri Jul 29 14:52:22 2016] [error] [client 155.47.45.34] TclError: no display name and no $DISPLAY environment variable, referer: http://lexos.wheatoncollege.edu/cluster
```

we tried *lots* of changes, but we cannot get the server to work with our current version of `matplotlib` calls;

note:  on the server, we don't have a display, so the warning (somehow) makes sense;

------------
we tried changing
1.  the Lexos.wsgi file to include
```
import os
os.environ['MPLCONFIGDIR'] = "/tmp/Lexos/"
os.environ['MPLBACKEND'] = "TkAgg"   # and tried "Agg" also
```
2. we uninstalled matplotlib and then installed it again with `conda install`

so, in short, we got close (again, most of the app was running fine on the server), but matplotlib again has reared its ugly head ...;  this is a tough admin issue for sure ...

**Comment by chantisnake** [comment]
try to change matplotlib backend?

**Comment by mleblanc321** [comment]
yes, yes (sorry);  this issue involves getting our new repo to run on the (old) Lexos server (Ubuntu v12, which is not the same as our dev boxes which were Ubuntu v14; however, it is not the OS in my opinion);  somehow, we need to find a setting on the server for matplotlib, perhaps via an Apache configuration and/or os.environ[] settings via the Python script as it starts;

there is _much_ discussion on StackOverflow about the issue, but Caleb and i were not able to find the magic mix;

**Comment by scottkleinman** [comment]
Apparently, this required

```python
import matplotlib
matplotlib.use('Agg') 
```

at the very start of `lexos.py`. Mark deserves a very large bottle of wine for all the blood, sweat, and tears it cost him figuring this out.

**Comment by mleblanc321** [comment]
:)
